cpg_workflows/jobs/gatk.py

```diff
@@ -0,0 +1,73 @@
+import hailtop.batch as hb
+from hailtop.batch.job import Job
+
+from cpg_utils import Path
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import command, fasta_res_group, image_path
+
+
+def preprocess_intervals(b, intervals_path, job_attrs, output_path):
+    j = b.new_job('Preprocess intervals', job_attrs)
+    j.image(image_path('gatk'))
+
+    sequencing_type = get_config()['workflow']['sequencing_type']
+    reference = fasta_res_group(b)
+
+    if sequencing_type == 'exome':
+        cmd = f"""
+        gatk PreprocessIntervals --reference {reference.base} --intervals {intervals_path} \\
+            --padding 250 --bin-length 0 --interval-merging-rule OVERLAPPING_ONLY \\
+            --output {j.preprocessed_intervals}
+        """
+
+    elif sequencing_type == 'genome':
+        cmd = f"""
+        gatk PreprocessIntervals --reference {reference.base} \\
+            --interval-merging-rule OVERLAPPING_ONLY --padding 0 \\
+            --output {j.preprocessed_intervals}
+        """
+
+    else:
+        raise ValueError('workflow/sequencing_type must be one of: "exome", "genome"')
+
+    j.command(command(cmd))
+    b.write_output(j.preprocessed_intervals, output_path)
+    return [j]
+
+
+def collect_read_counts(b, sample, intervals, job_attrs, output_path):
+    j = b.new_job(f'Collect gCNV counts for {sample.id}', job_attrs)
+    j.image(image_path('gatk'))
+
+    reference = fasta_res_group(b)
+    cmd = f"""
+    gatk CollectReadCounts \\
+      --reference {reference.base} --intervals {intervals} \\
+      --interval-merging-rule OVERLAPPING_ONLY \\
+      --input {sample.bam} --read-index {sample.bai} \\
```

Comment:
Clearly this line needs updating to how you actually get the CRAM and CRAI path out of a sample.

However I just came across this commentary in `_haplotype_caller_one` in _jobs/genotype.py_:

```python
# …can't pass GCS URL to CRAM to gatk directly because we will hit GCP egress bandwidth limit…
…
# Retrying copying to avoid google bandwidth limits
retry_gs_cp {str(cram_path.path)} $CRAM
retry_gs_cp {str(cram_path.index_path)} $CRAI
```

Uh oh! Perhaps this step will also need to make a temporary local copy of the CRAM file?

Reply:
That sounds reasonable!